Sources/JavaRuntime/AndroidSupport.cpp

```diff
@@ -0,0 +1,98 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift.org project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#ifdef __ANDROID__
```

Comment:
Comment from other thread @lhoward:

> This code needs to be removed and replaced with targeting a NDK version which exports the JNI.
> 
> It should be fairly trivial to do with a current Swift Android SDK, I just haven't had time to do it.

How about we do these steps to use the current android sdk right away?

Reply:
I'll try and take a look tomorrow.

Reply:
I will close this PR then. Thanks

Reply:
@lhoward Wouldnt the `JavaObjectHolder.swift` changes still be needed though?

Reply:
Ah I see what I did, I squashed removing `AndroidSupport.cpp` into the sample commit. Let me fix that now.

Reply:
OK, so my PR has three commits:

* ba56de0a03c84ef38a8ef0feb5a4efc200a92c6f which contains Android-specific changes to the top-level package
* 4d4d60e0fabe5d75b3817e64f03fb961f2e70b1e is the `JavaObjectHolder.swift` to which you refer. This is broken out separately as I believe it really needs to be applied on all platforms.
* 79d7d0d71298964ce04a9f56e501f29020efc6e9 is a sample Android application